canonicalwebteam/search/views.py

```diff
@@ -61,12 +61,18 @@ def search_view():
         results = None
 
         if query:
-            agent = user_agents.parse(str(flask.request.user_agent))
+            # Block weird characters
+            illegal_characters = ("【", "】")
+
+            if any(char in query for char in illegal_characters):
+                flask.abort(403, "Search query contains an illegal character")
 
-            # Block search bots
-            # So we don't pay for their API calls
+            # Block if a search bot
+            agent = user_agents.parse(str(flask.request.user_agent))
             if agent.is_bot:
-                flask.abort(403)
+                flask.abort(
+                    403, "Search engine crawlers can't perform searches"
```

Comment:
Should we be more specific about the error in case it's a genuine search? But I guess that will tip off the spammers...

Reply:
In this specific case they are presumably deliberately declaring themselves as a bot through the user-agent string. They shouldn't be surprised by this message.